pages/api/self-check/patch.ts

```diff
@@ -0,0 +1,25 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { SelfCheckGroup } from "../models/selfCheckGroup";
+
+const patchSelfCheckByID = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const { id } = req.query;
+    const selfCheck = await SelfCheckGroup.findById(id).exec();
+
+    if (!selfCheck)
+        return res
+            .status(404)
+            .send("The self check with the given ID was not found.");
+
+    for (const key in req.body) {
```

Comment:
You might be able to leverage javascript and do something like this instead: https://stackoverflow.com/a/51677542 (if you can get the raw dicts instead of the full object)